src/main/java/org/sagebionetworks/web/client/widget/upload/FileHandleUploadWidget.java

```diff
@@ -10,10 +11,16 @@
  *
  */
 public interface FileHandleUploadWidget extends IsWidget{
+	
+	public void reset();
 
-	/**
-	 * Configure the widget before using.
-	 * @param callback Will be called with the uploaded fileHandleId if the user successfully uploads a file.
-	 */
-	public void configure(String buttonText, CallbackP<String> callback);
+	FileMetadata[] getSelectedFileMetadata();
+
+	void configureValidation(FileValidator validator,

```

Comment:
What do you think about refactoring so that the validFileCallback is owned by the file validator (validator.setValidFileCallback), and renaming this method from "configureValidation" to just "setValidator"?